packages/kit/src/runtime/server/utils.js

```diff
@@ -32,16 +32,28 @@ export function decode_params(params) {
 	return params;
 }
 
+/** @param {any} obj **/
+function is_readable_node_stream(obj) {
+	// Copied from nodejs sources
+	return !!(
+		(
+			obj &&
+			typeof obj.pipe === 'function' &&
+			typeof obj.on === 'function' &&
+			(!obj._writableState || obj._readableState?.readable !== false) && // Duplex
+			(!obj._writableState || obj._readableState)
+		) // Writable has .pipe.
+	);
+}
```

Comment:
Thanks. We need to make this check as fast as possible (it's an edge case, we don't want to dwell on it) — we already know `obj` is truthy, so a sufficient check is probably that it has a `_readableState` property and (belt and braces, once we've established that) that `pipe` is a function:

```suggestion
	return obj._readableState && typeof obj.pipe === 'function';
}
```

Open question whether we need the indirection of a separate function call at that point

Reply:
Yeah you're right. I applied your suggestion and inlined the function.